### PR TITLE
GDScript: Add missing member type check when resolving `extends`

### LINKED ISSUE #75870: Declaring an inner class that extends a variable name crashes the engine, instead of just reporting an error (GDScript)
### Godot version

4.0 stable official

### System information

Windows 10

### Issue description

This has been posted by an user over at godotforums.org, in a more complicated form. I just made a super-minimal example that's trivial to reproduce. Don't know if this was already reported.

Engine crashes to desktop without a warning when following script is created and then saved:
~~~
var B = 1
class A extends B:
	pass
~~~

I also tried it in console version with verbose and debug flags enabled. The output to console was:
ERROR: FATAL: Condition "!exists" is true.
at: operator[] (./core/templates/hash_map.h:504) 

### Steps to reproduce

1. Create a new GDScript script file
2. Enter the following code
~~~
var B = 1
class A extends B:
	pass
~~~
3. Save the script file.
4. Engine crashes immediately upon save.

### Minimal reproduction project

N/A, trivial to reproduce.

### PR DESCRIPTION
**Pull request by dalexeev**
Closes #75870.

### CODE REVIEW

**Review by vnen — APPROVED** [review]

File: modules/gdscript/gdscript_analyzer.cpp [hunk 1]
```diff
@@ -506,6 +523,9 @@ Error GDScriptAnalyzer::resolve_class_inheritance(GDScriptParser::ClassNode *p_c
 			if (!id_type.is_set()) {
 				push_error(vformat(R"(Could not find nested type "%s".)", id->name), id);
 				return ERR_PARSE_ERROR;
+			} else if (id_type.kind != GDScriptParser::DataType::SCRIPT && id_type.kind != GDScriptParser::DataType::CLASS) {
```
Comment (dalexeev) [inline modules/gdscript/gdscript_analyzer.cpp:526]: I still have some doubts about this. We should probably check if the identifier is a class or a constant containing a script/class. Also, I am not happy with the variety of error messages:

* Constant "A" is not a preloaded script or class.
* Identifier "X" is not a preloaded script or class.
* Cannot get nested types for extension from non-GDScript type "RefCounted".
* Could not find nested type "Baz".
* Cannot use variable "A" in extends chain.

Comment (YuriSizov) [reply modules/gdscript/gdscript_analyzer.cpp:526]: Let's fix the crash for now, and iterate on this in a follow-up.

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3. This one slightly depends on https://github.com/godotengine/godot/pull/74844, but seems like usages of `id` can just be reverted to use `p_class`, which is what I did to fix up when picking.

**Pull request merged**